lib/galaxy/jobs/__init__.py

```diff
@@ -1151,9 +1155,10 @@ def _get_tool_evaluator(self, job):
         return tool_evaluator
 
     def _fix_output_permissions(self):
-        for path in [dp.real_path for dp in self.get_mutable_output_fnames()]:
-            if os.path.exists(path):
-                util.umask_fix_perms(path, self.app.config.umask, 0o666, self.app.config.gid)
+        if self.working_directory_exists():
```

Comment:
Nitpicking question: would it make sense to move this check to `fail()`, before calling this method? I *think* it's not needed when `_fix_output_permissions()` is called inside `finish()`.

Reply:
true, we can do that.